File: Company/src/main/webapp/service/fakturaService.js
/**
 * Created by JELENA on 20.6.2017.
 */

app.service('fakturaService', function($http){
    return {
        read: function(onSuccess, onError){
            $http.get('/api/fakture').then(onSuccess, onError);
        },
        readFaktura: function (id, onSuccess, onError) {
          $http.get('/api/fakture/' + id).then(onSuccess, onError);
        },
        readDobavljac: function (onSuccess, onError) {
            $http.get('api/fakture/firmaDobavljac/').then(onSuccess, onError);
        },
        readKupac: function (onSuccess, onError) {
            $http.get('api/fakture/firmaKupac').then(onSuccess, onError);
        },
        create: function(faktura, onSuccess, onError){
            $http.post('/api/fakture', faktura).then(onSuccess, onError);
        },
        update: function(faktura, onSuccess, onError){//mozda i nece trebati
            $http.patch('/api/fakture/' + faktura.id, faktura).then(onSuccess, onError);
        },
        generisiPdf: function(faktura, onSuccess, onError){
            $http.post('/api/fakture/generisiPdf', faktura).then(onSuccess, onError);
        }
    }
});

File: Company/src/main/java/com/ftn/service/FakturaService.java
package com.ftn.service;

import com.ftn.model.dto.FakturaDTO;

import java.util.List;

/**
 * Created by JELENA on 19.6.2017.
 */
public interface FakturaService {

    List<FakturaDTO> read();

    FakturaDTO readFaktura(Long id);

    List<FakturaDTO> readDobavljac();

    List<FakturaDTO> readKupac();

    FakturaDTO create(FakturaDTO fakturaDTO);

    FakturaDTO update(Long id, FakturaDTO fakturaDTO);

}
